src/compositemark/boxplot.ts

```diff
@@ -309,6 +313,16 @@ export function normalizeBoxPlot(
   };
 }
 
+function extractDependencyTransformsOfGroupby(transforms: Transform[], groupby: string[]): Transform[] {
```

Comment:
What is a dependency transform?

Reply:
With this spec
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
  "data": {
    "url": "data/cars.json"
  },
  "mark": "boxplot",
  "encoding": {
    "y": {
      "field": "Miles_per_Gallon",
      "type": "quantitative"
    },
    "x": {
      "field": "Year",
      "type": "temporal",
      "timeUnit": "year"
    }
  }
}
```
Vega-Lite does not add timeUnit transform `{"timeUnit": "year", "field": "Year", "as": "year_Year"}` in to the `transform` for `filteredLayers`, which includes outliers and whiskers or boxplot. However, the joinaggregate transform in the `filteredLayers` has groupby `['year_Year']`. So the joinaggregate is depending on the timeUnit transform.

This is the normalized spec before including the timeUnit transform to the `filteredLayers`
```
{
  "$schema": "https://vega.github.io/schema/vega-lite/v3.json",
  "data": {"url": "data/cars.json"},
  "layer": [
    {
      "transform": [
        {
          "joinaggregate": [
            {
              "op": "q1",
              "field": "Miles_per_Gallon",
              "as": "lower_box_Miles_per_Gallon"
            },
            {
              "op": "q3",
              "field": "Miles_per_Gallon",
              "as": "upper_box_Miles_per_Gallon"
            }
          ],
          "groupby": ["year_Year"]
        }
      ],
      "layer": [
        {
          "transform": [
            {
              "filter": "(datum.Miles_per_Gallon < datum.lower_box_Miles_per_Gallon - 1.5 * (datum.upper_box_Miles_per_Gallon - datum.lower_box_Miles_per_Gallon)) || (datum.Miles_per_Gallon > datum.upper_box_Miles_per_Gallon + 1.5 * (datum.upper_box_Miles_per_Gallon - datum.lower_box_Miles_per_Gallon))"
            }
          ],
          "mark": {"type": "point", "style": "boxplot-outliers"},
          "encoding": {
            "y": {"field": "Miles_per_Gallon", "type": "quantitative"},
            "x": {
              "title": "Year (year)",
              "type": "temporal",
              "field": "year_Year",
              "axis": {"format": "%Y"}
            }
          }
        },
        {
          ... whiskers of boxplot
        }
      ]
    },
    {
      ... box and mid tick of the boxplot
    }
  ]
}
```

Reply:
Instead of extracting it here, which is a hacky solution, better refactor `boxParams` to provide this kind of information from the beginning.